src/test/util/event.rs

```diff
@@ -500,15 +529,52 @@ impl EventSubscriber<'_> {
         .flatten()
     }
 
-    pub(crate) async fn collect_events<F>(&mut self, timeout: Duration, mut filter: F) -> Vec<Event>
-    where
-        F: FnMut(&Event) -> bool,
-    {
-        let mut events = Vec::new();
-        while let Some(event) = self.wait_for_event(timeout, &mut filter).await {
-            events.push(event);
-        }
-        events
+    /// Waits for the next CommandStartedEvent/CommandFailedEvent pair.
+    /// If the next CommandStartedEvent is associated with a CommandFailedEvent, this method will
+    /// panic.
+    pub(crate) async fn wait_for_successful_command_execution(
```

Comment:
This functionality existed on `EventClient`, but I wanted to use a regular `Client` with a handler, so I reimplemented it here too.